source/Octopus.Client/Model/DeploymentStepResource.cs

```diff
@@ -169,6 +169,13 @@ public DeploymentActionResource AddOrUpdateScriptAction(string name, ScriptActio
         public DeploymentActionResource AddOrUpdatePackageAction(string name, PackageResource package)
         {
             var action = AddOrUpdateAction(name);
+            var packageReference = new PackageReference
+            {
+                FeedId = package.FeedId,
+                PackageId = package.PackageId
+            };
+            action.Packages.Clear();
```

Comment:
Good question. I added the `Clear` to prevent multiple package references if that method was run multiple times but for different packages.
Digger deeper into the consequences of this:
The `TentaclePackage` step type can only operate on a single package. There is validation in place to prevent adding "named" packages:
![image](https://user-images.githubusercontent.com/171197/93402361-ec9b4300-f8c7-11ea-90fa-429c5afa7868.png)
Package references are unique by both name and id. There can't be two un-named packages in the package collection:
![image](https://user-images.githubusercontent.com/171197/93402484-32580b80-f8c8-11ea-811f-43aec5beafd8.png)

Sooo, the constraints on the package collection and on the action type mean there can be only one. I think we're ok.